playbooks/deploy-ocp-sno.yml

```diff
@@ -0,0 +1,566 @@
+      ansible.builtin.import_role:
+        name: redhatci.ocp.setup_http_store
+
+    - name: Deploy or redeploy the OpenShift client
+      vars:
+        oc_client_install_url: "{{ ocp_version_facts_oc_client_pull_link }}"
+        oc_client_install_archive_dest_dir: "{{ dest_iso_dir }}"
+        oc_clinet_install_version: "{{ ocp_version_facts_parsed_release }}"
+      ansible.builtin.import_role:
+        name: oc_client_install
+
+    - name: Download and extract the OpenShift installer
+      vars:
+        openshift_version: "{{ ocp_version_facts_parsed_release }}"
+        release_image: "{{ ocp_version_facts_pull_spec }}"
+        extract_dest_path: "{{ openshift_installer_extract_dest_path }}"
+      ansible.builtin.import_role:
+        name: redhatci.ocp.extract_openshift_installer
+
+    - name: Mirror OpenShift release (disconnected)
+      when: disconnected | bool
+      vars:
+        mor_oc: "{{ ansible_user_dir }}/.local/bin/oc"
+        mor_version: "{{ ocp_version_facts_parsed_release }}"
+        mor_pull_url: "{{ ocp_version_facts_pull_spec }}"
+        mor_cache_dir: "/opt/cache"
+        mor_webserver_url: "http://{{ ansible_host }}:{{ share_http_iso_port }}/"
+        mor_registry_url: "{{ disconnected_registry_url }}:{{ disconnected_registry_port }}"
+        mor_registry_path: "ocp4/openshift"
+        mor_auths_file: "{{ pull_secret_file }}"
+        mor_force: true
+        mor_install_type: "abi"
+        mor_is_type: "idms"
+        mor_mirror_disk_images: true
+        mor_mirror_container_images: true
+        mor_write_custom_config: true
+      ansible.builtin.include_role:
+        name: redhatci.ocp.mirror_ocp_release
+
+- name: Render masters variables
+  hosts: masters
+  gather_facts: false
+  vars:
+    ipv6_only: false
+    ipv4_only: false
+    disconnected: false
+  tasks:
+
+    - name: Render masters variables
+      delegate_to: bastion
+      ansible.builtin.set_fact:
+        vm_spec: "{{ vm_spec_string | from_yaml }}"
+        network_interfaces: "{{ network_interfaces_string | from_yaml }}"
+        network_config: "{{ network_config_string | from_yaml }}"
+        mac_interface_map: "{{ mac_interface_map_string | from_yaml }}"
+
+
+    - name: Remove IPv4 default route for dual-stack (disconnected)
+      delegate_to: bastion
+      when: disconnected | bool
+      ansible.builtin.set_fact:
+        network_config: >-
+          {{
+            network_config | combine({
+              'raw': network_config.raw | combine({
+                'routes': {
+                  'config': (
+                    network_config.raw.routes.config
+                      | default([])
+                      | rejectattr('destination', 'equalto', '0.0.0.0/0')
+                      | list
+                  )
+                }
+              })
+            }, recursive=True)
+          }}
+
+    - name: Replace DNS servers with internal registry; set routes to internal CIDR blocks (disconnected)
```

Comment:
note - this config will  remove all internet access from the host 
just make sure thats the intend result